### ISSUE TITLE
Poetry does not respect the `develop = false` option when installing from path

### ISSUE THREAD

**Issue by lejafar** [python-poetry/poetry#1451]
- [x] I am on the [latest](https://github.com/sdispater/poetry/releases/latest) Poetry version.
- [x] I have searched the [issues](https://github.com/sdispater/poetry/issues) of this repo and believe that this is not a duplicate.

- **OS version and name**: MacOS Majove
- **Poetry version**: 0.12.17
- **Link of a [Gist](https://gist.github.com/lejafar/5c5f13742b3a600640bdeb3b7cbbebfa)**

## Issue
So when I run `poetry install` on the `pyproject.toml` file linked in the gist, it will install the package `bar` specified as a `path` in develop mode, even though the `develop` flag was set to false

```
$ poetry run pip list
Package    Version Location
---------- ------- ------------------------------------
bar      1.0.0   /private/tmp/foo/submodules/bar
```

**Comment by gvskalyan** [comment]
Hi, I was also encountering the same problem,
1.  poetry install on a local custom package even with develop=false, produces an egg link in the virtual environment's site-packages which has an absolute path as shown above in location.
2. Due to this egg-link problem when I add another project as dependency using `--path` both the projects get egg-links in site-packages with their respective absolute paths, which prevents us from using poetry to generate build and use it on different machine/path.

**Comment by Morikko** [comment]
Hi, I added a package with a `package_dir` and a `packages` fields in the `setup.py`.
Due to the egg link that points to the path, the import is working on the `package_dir` without filtering on the `packages`.

Thus, I was unable to test if the import was really working as other libs I didn't want in the packaged target could be available. Finally, I had to drop poetry even if it does a great job for everything else.

Note: I know this package setup could be a code smell. But, I am working on old code I am migrating slowly. So, I have to deal with it for some time.

**Comment by petergaultney** [comment]
I think there's some confusion within `poetry` about what the `develop` flag is even intended for.

In `pip`, these are generally considered "editable" requirements, which is why in `pipenv` the flag for marking them with "-e " is `editable`, not `develop`.

I would welcome some clarity around what behavior `poetry` intends for editable and non-editable path requirements.

**Comment by lejafar** [comment]
The problem seems to be resolved in `1.0.0` 💪
